### PR TITLE
auto join twitter-sync-state libraries from twittersocialgraph

### PR DESCRIPTION
**Pull request by ahsu1230**
@andrewconner

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/common/social/TwitterSocialGraph.scala [hunk 1]
```diff
@@ -168,6 +172,27 @@ class TwitterSocialGraphImpl @Inject() (
     } yield {
       val mutualFollows = followerIds.toSet.intersect(friendIds.toSet)
       log.info(s"[fetchSocialUserInfo(${socialUserInfo.socialId})] friendIds(len=${friendIds.length}):${friendIds.take(10)} followerIds(len=${followerIds.length}):${followerIds.take(10)} mutual(len=${mutualFollows.size}):${mutualFollows.take(10)}")
+
+      SafeFuture {
+        db.readOnlyMaster { implicit s =>
+          log.info(s"[fetchSocialUserInfo(${socialUserInfo.socialId})] fetching twitter_syncs for ${friendIds.length} friends...")
+          //twitterSyncStateRepo.getTwitterSyncsByFriendIds(friendIds.map(Id[User](_)).toSet)
+          Map.empty[Id[User], TwitterSyncState]
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/TwitterSocialGraph.scala]: :)

Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/TwitterSocialGraph.scala]: woops was checking something lol (related to the execution context thing)

File: kifi-backend/modules/shoebox/app/com/keepit/common/social/TwitterSocialGraph.scala [hunk 2]
```diff
@@ -168,6 +172,27 @@ class TwitterSocialGraphImpl @Inject() (
     } yield {
       val mutualFollows = followerIds.toSet.intersect(friendIds.toSet)
       log.info(s"[fetchSocialUserInfo(${socialUserInfo.socialId})] friendIds(len=${friendIds.length}):${friendIds.take(10)} followerIds(len=${followerIds.length}):${followerIds.take(10)} mutual(len=${mutualFollows.size}):${mutualFollows.take(10)}")
+
+      SafeFuture {
+        db.readOnlyMaster { implicit s =>
+          log.info(s"[fetchSocialUserInfo(${socialUserInfo.socialId})] fetching twitter_syncs for ${friendIds.length} friends...")
+          //twitterSyncStateRepo.getTwitterSyncsByFriendIds(friendIds.map(Id[User](_)).toSet)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/TwitterSocialGraph.scala]: This could be huge, let's group it by batches of 100 or so.

**Comment by andrewconner** [comment]
Awesome, thanks

File: kifi-backend/modules/shoebox/app/com/keepit/common/social/TwitterSocialGraph.scala [hunk 3]
```diff
@@ -168,6 +172,27 @@ class TwitterSocialGraphImpl @Inject() (
     } yield {
       val mutualFollows = followerIds.toSet.intersect(friendIds.toSet)
       log.info(s"[fetchSocialUserInfo(${socialUserInfo.socialId})] friendIds(len=${friendIds.length}):${friendIds.take(10)} followerIds(len=${followerIds.length}):${followerIds.take(10)} mutual(len=${mutualFollows.size}):${mutualFollows.take(10)}")
+
+      SafeFuture {
+        db.readOnlyMaster { implicit s =>
+          log.info(s"[fetchSocialUserInfo(${socialUserInfo.socialId})] fetching twitter_syncs for ${friendIds.length} friends...")
+          twitterSyncStateRepo.getTwitterSyncsByFriendIds(friendIds.map(Id[User](_)).toSet)
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/common/social/TwitterSocialGraph.scala:179]: Sorry, I wasn't clear. This should be grouped. Imagine `friendIds` may be very large (I follow 2500 people. Or 25,000)

**Pull request merged**